cmd/common.go

```diff
@@ -614,3 +631,116 @@ func exportCodesignFiles(tool Tool, archivePath, outputDirPath string) error {
 
 	return nil
 }
+
+func askUploadProvProfiles() (bool, error) {
+	messageToAsk := "Do you want to upload the provisioning profiles to Bitrise?"
+
+	answer, err := goinp.AskForBoolFromReader(messageToAsk, os.Stdin)
+	if err != nil {
+		return answer, err
+	}
+
+	return answer, nil
+}
+
+// ----------------------------------------------------------------
+// --- Upload methods
+
+// TODO
+func uploadProvisioningProfiles(profilesToUpload []profileutil.ProvisioningProfileInfoModel, outputDirPath string) error {
+	// Get accesToken from stdin
+	accessToken, err := uploaders.AskAccessToken()
+
+	appList, err := uploaders.FetchMyApps(accessToken)
+	if err != nil {
+		log.Errorf("Failed to ask your app list from Bitrise: %s", err)
+		return err
+	}
+
+	selectedApp, err := selectApp(appList)
+	if err != nil {
+		return err
+	}
+
+	// Upload provisioning profiles
+	for _, profile := range profilesToUpload {
+
+		exportFileName := provProfileExportFileName(profile, outputDirPath)
+
+		provProfile, err := os.Open(outputDirPath + "/" + exportFileName)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := provProfile.Close(); err != nil {
+				log.Warnf("Provisioning profile close failed, err: %s", err)
+			}
+
+		}()
+
+		info, err := provProfile.Stat()
+		if err != nil {
+			return err
+		}
+
+		bytes := info.Size()
+		log.Debugf("\n%s size: %d", exportFileName, bytes)
+
+		provProfSlugResponseData, err := uploaders.RegisterProvisioningProfile(accessToken, selectedApp.Slug, bytes, profile)
+		if err != nil {
+			return err
+		}
+
+		err = uploaders.UploadProvisioningProfile(provProfSlugResponseData, outputDirPath, exportFileName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func selectApp(appList []uploaders.Appliocation) (seledtedApp uploaders.Appliocation, err error) {
+	selectionList := []string{}
+
+	for _, app := range appList {
+		selectionList = append(selectionList, app.Title+" ("+app.RepoURL+")")
+	}
+
+	userSelection, err := goinp.SelectFromStringsWithDefault("Select the app which you want to upload the privisioning profiles", 1, selectionList)
+
+	if err != nil {
+		return uploaders.Appliocation{}, fmt.Errorf("failed to read input: %s", err)
+
+	}
+
+	log.Debugf("selected app: %v", userSelection)
+
+	selectedApp, err := getAppFromUserSelection(userSelection, appList)
```

Comment:
no need for this function:

- sort the selectionList
- then find the selected index by: sort.Search...
- then appList[i] is the app

Reply:
I have changed to the sort.Search() method as you advised, but I have not removed the getAppFromUserSelection() because it has tests.